### DISCUSSION TITLE
Client.disconnect() from the client side if there i no internet

### DISCUSSION THREAD

**Discussion by afahim12** [socketio/socket.io#4151] [Q&A] [answered]
If there is no network connection, does Client.disconnect() from the client side (not server side) work? or do we need some sort of server handshake  for the disconnection to really fire? Thank you! Ahmed

**Comment by darrachequesne** [comment, marked as answer]
Yes, it works even without a server.

**Comment by afahim12** [comment]
Merci Damien!

And also thanks for your contributions on socket 4 and redis.
Just implemented it and used fetch sockets across all node/dyno servers on heroku and it worked like a charm. 

Very thankful! 
______________________
Ahmed Fahim
+1 (917) 443-9850
***@***.***

**Reply by darrachequesne** [reply]
Thanks for the feedback, that really means a lot :heart:
